src/o365/spo/commands/site/site-set.ts

```diff
@@ -301,6 +307,52 @@ class SpoSiteSetCommand extends SpoCommand {
     return Utils.executeCommand(spoSiteDesignApplyCommand as Command, options, cmd);
   }
 
+  private setsharingCapabilities(cmd: CommandInstance, args: CommandArgs): Promise<void> {
+    if (typeof args.options.sharingCapabilities === 'undefined') {
+      return Promise.resolve();
+    }
+
+    return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
+      if (this.verbose) {
+        cmd.log(`Retrieving request digest...`);
+      }
+
+      let sharingCapability: SharingCapabilities = SharingCapabilities[(args.options.sharingCapabilities as keyof typeof SharingCapabilities)];
+     
+      this.getSpoAdminUrl(cmd,this.debug)
+      .then((_spoAdminUrl: string): Promise<ContextInfo> => {
+        this.spoAdminUrl = _spoAdminUrl;
+        return this.getRequestDigest(this.spoAdminUrl);
+      }).then((res: ContextInfo): Promise<string> => {
+        if (this.verbose) {
+          cmd.log(`Setting sharing for site  ${args.options.url} as ${args.options.sharingCapabilities}`);
+        }
+
+        const requestOptions: any = {
+          url: `${this.spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+          headers: {
+            'X-RequestDigest': res.FormDigestValue
+          },
+          body: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="2" ObjectPathId="1"/><ObjectPath Id="4" ObjectPathId="3"/><SetProperty Id="5" ObjectPathId="3" Name="SharingCapability"><Parameter Type="Enum">${sharingCapability}</Parameter></SetProperty><ObjectPath Id="7" ObjectPathId="6"/><ObjectIdentityQuery Id="8" ObjectPathId="3"/></Actions><ObjectPaths><Constructor Id="1" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}"/><Method Id="3" ParentId="1" Name="GetSitePropertiesByUrl"><Parameters><Parameter Type="String">${args.options.url}</Parameter><Parameter Type="Boolean">false</Parameter></Parameters></Method><Method Id="6" ParentId="3" Name="Update"/></ObjectPaths></Request>`
```

Comment:
Since site's URL is provided by the user, we should XML-escape it before using in the query